### ISSUE TITLE
Lagging with Android Animations using Lottie Json at same time

### ISSUE THREAD

**Issue by jonasdoliveira** [airbnb/lottie-android#337]
Hello,

I've facing some performance problems trying to use Lottie in a screen that also has any Android animation. Like open an Options Menu on ActionBar.
I had tested a lot of sample files from lottie-android repository, but many of them was easy to see this lag/delay.
For example using the sample for example:
https://github.com/airbnb/lottie-android/blob/master/LottieSample/src/main/assets/lottiefiles.com%20-%20VR%20Sickness.json

It is possible to observe that has an delay to open this menu OptionMenu, it is related to the Android's animation.

Also it is possible to observe that the FPS on screen has reduced to about 30 fps and when the animation ends, it returns to 60fps. Even with animations created with 60fps.
Also the CPU increases when running the Lottie animation.

I believe that this problems is related to concurrence with UI Thread of Android, that tries to reproduce all the animations.

Is there anything that can be done to a better performance? Is that a known issue?

**Comment by gpeal** [comment]
What device are you using? I'm unable to reproduce inconsistent performance on any of mine.

Regarding overall performance or cpu usage, complex animations will definitely require cpu and may drop below 60fps on some phones but there isn't a ton that can be done about that.

**Comment by jonasdoliveira** [comment]
Hi @gpeal!
I tested in many different devices Sony and Motorola for example, new devices and old ones. All have similar behavior. Of course that newer ones were faster but still has the lag.

I'm attaching a simple .apk using the same animation mentioned before mentioned. I tested with bigger animations (about 300kb) and small ones (16kb), With small animations everything worked fine and there is no lag.

Please check the attached apk.
There is an animation of Lottie running. Touch on the Options Menu at top right corner to see the delay during the creation of menu options dialog.
You can also check the FPS by a balloon on the screen that I added.
Also you can stop or start the animation again, try it with animation running and not. Observe the delay during animation of menu and also the FPS that changes a lot.

If there is anything else that I can help, please let me know!

[testLottieLag.apk.zip](https://github.com/airbnb/lottie-android/files/1071992/testLottieLag.apk.zip)

**Comment by jonasdoliveira** [comment]
I have done another test with an application that I found on internet that shows all the native Android animations.

Doing the same thing by try to open the Options Menu it is possible to see the delay also using only native Android animations.

I'm guessing that it is an Android problem. It is not able to handle to many process of animations at same time, since it only has one thread for UI.

Attached is the test apk with Android Native animations.
[app-debug.apk.zip](https://github.com/airbnb/lottie-android/files/1075684/app-debug.apk.zip)

**Comment by jonasdoliveira** [comment]
@gpeal, have you checked anything related to that issue? Thanks!

**Comment by gpeal** [comment]
@jonasdoliveira I'm don't think there is anything actionable here. I'm continuing to work on general performance improvements, however.

**Comment by jonasdoliveira** [comment]
@gpeal ok! Did you see the problem thought the apps I've created? I think it is an Android problem that is not able yet to handle multiple animations at same screen at same time. :(
If you see any news please let me know, I really appreciate the Lottie!

**Comment by gpeal** [comment]
@jonasdoliveira this actually doesn't reproduce on my device. I'm not sure if there is anything I can do on the lottie side unfortunately. You can try running systrace with the latest version of lottie on master with https://github.com/airbnb/lottie-android/blob/master/lottie/src/main/java/com/airbnb/lottie/LottieAnimationView.java#L567 enabled, then you can run systrace with `--app=YOUR_PACKAGE` to get more info

**Comment by Fereshtehnaji** [comment]
hello , I have the same problem. I use Lottie and some other animation, but i see lag on every animation on my app, even opening navigation bar.

**Comment by dragonjia** [comment]
I'm interesting about this topic , has the  problem be solved eventually? @jonasdoliveira

**Comment by gpeal** [comment]
@Fereshtehnaji @dragonjia Mattes and masks incur a large performance overhead proportional to their size. There is some more [documentation here](http://airbnb.io/lottie/android/performance.html)

**Comment by pagrawal1407** [comment]
@gpeal I am seeing the same issue when I am trying to play several Lottie animations on the same Dialog screen (Each Lottie file is ~100KB).

PS: Above link of "documentation here" is not working anymore. It gives a 404 not found.

**Comment by gpeal** [comment]
@pagrawal1407 http://airbnb.io/lottie/#/android?id=performance
Make sure you're on 3.0 and try hardware acceleration (setRenderMode)

**Comment by pagrawal1407** [comment]
Thanks @gpeal for your help. I tried playing with hardware acceleration but didn't get much success. But my requirement didn't mandate all the animations to play at the same time, so I solved the issue by turning of autoplay animations. So, I didn't dive too deep into it.
I had another question. If I am downloading the json lottie file from network as a string, can I directly set it into the setAnimation method or do I need to convert it into some kind of Json object first?
